### PR TITLE
feat: support gleam lang

### LINKED ISSUE #26744: Support for gleam lang manager
As discussed in https://github.com/renovatebot/renovate/discussions/26717

### PR DESCRIPTION
**Pull request by SnakeDoc**
## Changes

Complete work started by @cdaringe and add gleam-lang manager support.

## Context

See #26767 and #26744 
Additional work addresses Code Review comments on PR #26767, and adds `rangeStrategy` support. A new review is necessary.

- closes #26744
## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

Note: The manager has been documented, and the previous PR signaled documentation was updated as well. If any additional documentation is missing/necessary, please point me in the right direction.

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/manager/gleam/index.ts
Comment (rarkins) [inline lib/modules/manager/gleam/index.ts:12]: Are you sure it can never be in a subdirectory?

File: lib/modules/manager/gleam/extract.ts
Comment (rarkins) [inline lib/modules/manager/gleam/extract.ts:1]: No parsing of manifest.toml to get the locked version?

File: lib/modules/manager/gleam/artifacts.ts
Comment (rarkins) [inline lib/modules/manager/gleam/artifacts.ts:32]: I don't see how this can ever be reached if packageFileName is valid

File: lib/modules/manager/gleam/artifacts.ts
Comment (rarkins) [inline lib/modules/manager/gleam/artifacts.ts:61]: Please separate these two conditions and add a debug message for each

**Review by rarkins — CHANGES_REQUESTED** [review]

File: lib/modules/manager/gleam/index.ts
Comment (SnakeDoc) [reply lib/modules/manager/gleam/index.ts]: it looks like this was discussed previously in [#26767](https://github.com/renovatebot/renovate/pull/26767#discussion_r1460622829). while gleam.toml can only ever be in the project's root dir, the existing regex did not evaluate gleam monorepo's or nested projects. i don't see a reason why renovate should limit that here, so this has been fixed in [636b682]. (https://github.com/renovatebot/renovate/pull/30345/commits/636b682583bf098770f33b393f08a650baf393e5)

File: lib/modules/manager/gleam/extract.ts
Comment (SnakeDoc) [reply lib/modules/manager/gleam/extract.ts:1]: i don't think this it is necessary, since the manager uses `gleam` itself to drive the lockfile (manifest.toml). so renovate does not need to understand or evaluate it as well. please let me know if i misunderstood your comment.

**Comment by SnakeDoc** [comment]
I see when PR's are eventually merged, they are squashed. I'd like to ensure @cdaringe get's proper credit for his significant work towards adding gleam support. He got us 95% of the way here.

File: lib/modules/manager/gleam/range.ts
Comment (rarkins) [inline lib/modules/manager/gleam/range.ts:25]: This range strategy is not possible unless you parse the lock file and include `lockedVersion` in the extract phase

**Review by rarkins — CHANGES_REQUESTED** [review]
Either drop update-lockfile support (and document it), or extract lockedVersion and add updateLockedDependency() support

**Review by rarkins — DISMISSED** [review]

**Comment by rarkins** [comment]
Please stop pushing commits if you don't have code changes

**Comment by SnakeDoc** [comment]
> Please stop pushing commits if you don't have code changes

my mistake. i had assumed you needed/wanted the branch up to date before final review.

**Comment by rarkins** [comment]
We don't need it because the github merge queue will essentially rebase it on top of main before running final checks before merge, and in this case it keeps resetting the test approvals which require a repo admin to approve each commit (for new contributors - this is a github policy/limitation)

**Review by rarkins — APPROVED** [review]

**Review by secustor — APPROVED** [review]

**Comment by SnakeDoc** [comment]
Wonderful! I believe this Closes #26744

**Pull request merged**
